### ISSUE TITLE
Is 100 - Prec@1 (EMA) the result in the paper?

### ISSUE THREAD

**Issue by prlz77** [ahmetius/LP-DeepSSL#4]
Hi! Nice work! 

I wanted to reproduce CIFAR100 + your method + mean teacher, 4000 labelled samples. I just run your code and I am getting this in the stage 2:

 * Prec@1 56.080        Prec@5 84.920
Evaluating the EMA model:
 * Prec@1 67.520        Prec@5 91.080

However I don't know how to interpret these in the context of the paper. Could you help me please?

**Comment by prlz77** [comment]
Also, the two commands to run your model with mean teacher in the README are the same, is it correct?

**Comment by zijian-hu** [comment]
It seems like
`Prec@1 56.080` means `top 1 accuracy is 56.08%`
`Prec@5 84.920` means `top 5 accuracy is 84.92%`

**Comment by prlz77** [comment]
Yes, I get this, but the EMA model gets way more than in their results. They report 43% error in the paper, here it is 33%

**Comment by ahmetius** [comment]
Hi,

I don't have access to the logs that were created when writing the paper, but I'm re-running the experiments and will get back to you shortly.

> Also, the two commands to run your model with mean teacher in the README are the same, is it correct?

Two commands are different. The command for the first stage calls `train_stage1.py` and the command for the second stage calls `train_stage2.py`.

**Comment by prlz77** [comment]
Thanks for the answer.

> Two commands are different. The command for the first stage calls train_stage1.py and the command for the second stage calls train_stage2.py.

I meant the arguments. Without mean teacher, the arguments are different for the two stages, while with mean teacher they are the same.

**Comment by prlz77** [comment]
@ahmetius the accuracies you report are the ones of "Evaluating the EMA model"?

**Comment by ahmetius** [comment]
> Thanks for the answer.
> 
> > Two commands are different. The command for the first stage calls train_stage1.py and the command for the second stage calls train_stage2.py.
> 
> I meant the arguments. Without mean teacher, the arguments are different for the two stages, while with mean teacher they are the same.

That's expected. The first stage w/o mean teacher does not make use of any unlabeled data. It is the simple supervised learning baseline. That's why it has different arguments.

**Comment by ahmetius** [comment]
> @ahmetius the accuracies you report are the ones of "Evaluating the EMA model"?

Yes, but the primary and EMA models shouldn't have a big difference after the training ends. Can you please write the exact commands which you have used. I ran the following two commands, and I get expected results:

**Command:**
`
python train_stage1.py --labeled-batch-size=31 --num-labeled=4000 --gpu-id=0 --label-split=10 --isMT=True --isL2=False --dataset=cifar100
`

**Output:**
Evaluating the primary model:
  Prec@1 54.340        Prec@5 80.160
Evaluating the EMA model:
  Prec@1 54.730        Prec@5 80.520
--- checkpoint saved to models/cifar100_4000_mean_teacher_split_10_isL2_0/checkpoint.180.ckpt ---
**Command:**
`
python train_stage2.py --labeled-batch-size=31 --num-labeled=4000 --gpu-id=0 --label-split=10 --isMT=True --isL2=False --dataset=cifar100
`

**Output:**
Evaluating the primary model:
 * Prec@1 56.060        Prec@5 82.560
Evaluating the EMA model:
 * Prec@1 56.490        Prec@5 83.000
--- checkpoint saved to models/cifar100_4000_mt_ss_split_10_isL2_0/checkpoint.180.ckpt ---

**Comment by prlz77** [comment]
It is the same command no? 
`python train_stage1.py --labeled-batch-size=31 --num-labeled=4000 --gpu-id=6 --label-split=10 --isMT=True --isL2=False --dataset=cifar100`
`python train_stage2.py --labeled-batch-size=31 --num-labeled=4000 --gpu-id=6 --label-split=10 --isMT=True --isL2=False --dataset=cifar100`
Let me rerun again

**Comment by prlz77** [comment]
It looks like it could be the pytorch version, I am checking it. I know it is tiring but maybe you should consider updating to pytorch 1.5, or provide a Docker. I have updated the code to work with 1.5 and I am trying it now, I can do a pull request if you wish (and if it works).
